packages/vite/src/node/server/index.ts

```diff
@@ -456,7 +456,7 @@ export async function _createServer(
     ? (chokidar.watch(
         // config file dependencies and env file might be outside of root
         [
-          root,
+          searchForWorkspaceRoot(config.root),
           ...config.configFileDependencies,
           ...getEnvFilesForMode(config.mode, config.envDir),
```

Comment:
I found that `pnpm test-serve /css/` passes if I comment out these lines. Maybe there's a bug here 🤔 But even with that running the whole test (`pnpm test-serve`) takes a long time. I guess it's because we have many files under the repository.

Reply:
I'm worried about the performance impact of watching the whole fs root. What I'm thinking now is that we may need to be more fine-grained. Maybe we could automatically watch parent folders in a shallow way (not deep watch) of every imported file for example.
For now, I think we should revert.